src/InstrumentationEngine/MethodInfo.cpp

```diff
@@ -1904,11 +1933,22 @@ HRESULT MicrosoftInstrumentationEngine::CMethodInfo::GetInstrumentationResults(
     {
         pMethodBody = (LPCBYTE) pMethodHeader + sizeof (IMAGE_COR_ILMETHOD_TINY);
         codeSize = ((COR_ILMETHOD_TINY*)pMethodHeader)->GetCodeSize();
+
+        //If a custom buffer was used by the raw profiler, we will not have the size of the
+        //final instrumentation memory. Attempt to compute it.
+        if (cbMethodSize == 0)
+        {
+            cbMethodSize = codeSize + sizeof(IMAGE_COR_ILMETHOD_TINY);
+        }
     }
     else
     {
         pMethodBody = (LPCBYTE)pMethodHeader + sizeof(IMAGE_COR_ILMETHOD_FAT);
         codeSize = ((COR_ILMETHOD_FAT*)pMethodHeader)->GetCodeSize();
+        if (cbMethodSize == 0)
+        {
+            cbMethodSize = codeSize + sizeof(COR_ILMETHOD_FAT);
```

Comment:
Need to actually get the size of the header since it can have additional sections for exceptions